src/cobra/core/gene.py

```
@@ -135,6 +183,138 @@ def visit_BinOp(self, node):
             raise TypeError("unsupported operation '%s'" % node.op.__class__.__name__)
 
 
+# Using unescaped ":", "," or " " causes sympy to make a range or separate items
+str_for_sympy_symbols_dict = {':': r'\:', ',': r'\,', ' ': r'\ '}
+
+
+def fix_str_for_symbols(unescaped_str):
```

better than opaque substitutions, rather raise an error if the IDs can't be converted to sympy and let the user fix the IDs

---

Hmm. This is more like escapeID for SBML, where that is done to correct the SBML export, but not raised as an error.